### DISCUSSION TITLE
How to have a native built app (containerised) pick up config resources (files) at runtime?

### DISCUSSION THREAD

**Discussion by hartmut-co-uk** [quarkusio/quarkus#33616] [Q&A]
Hi, I was wondering if file resources, such as property/conf files that are usually placed under `src/main/resources` can be dynamically provided, e.g. for a particular stage (dev/test/prod/...).

Example: Using _datastax/cassandra-quarkus_ can I mount a custom `application.conf` for the datastax java driver into my native app container to be picked up at runtime?

The key config is possible via `application.properties` or ENV vars, but e.g. advanced driver configs (ref https://quarkus.io/guides/cassandra#advanced-driver-configuration) such as [execution profiles](https://docs.datastax.com/en/developer/java-driver/4.15/manual/core/configuration/#execution-profiles) can't be configured that way. 

Is this correct? Does anyone have suggestions, best practices or other patterns on how to do this?

**Comment by bulldog98** [comment]
If you want to run in kubernetes you can use the [kubernetes-config](https://quarkus.io/guides/kubernetes-config) extension and configure your application to use a configmap/secret in production mode to read your config from.

**Reply by hartmut-co-uk** [reply]
Yes, thanks!
My question was, if the native app will actually pick up the file?
Also where would the config need to be placed?
Also I'm not sure if I can mount it directly to the container workdir `/work/` 🤔  (or even where it has to be placed to be picked up...)

**Reply by hartmut-co-uk** [reply]
Ignore parts of my previous comment, I'll do more reading first :-)

**Reply by hartmut-co-uk** [reply]
I found answers to some of my own questions, e.g. where I could mount / create a config file inside the container / workdir to be picked up... 
=> it should go to `$PWD/config/application.properties`
ref: https://quarkus.io/guides/config-reference#application-properties-file

I'm going to try to verify if other files (the datastax cassandra java client config file `application.conf` in my case) would also be recognised if placed in `$PWD/config/` path.
